lib/network/http_middlewares.go

```diff
@@ -86,9 +90,18 @@ func RateLimitMiddleware(logger logging.Logger, rule common.RateLimitRule) mux.M
 				stdlib.WithLimitReachedHandler(rateLimitReachedHandler),
 			)
 		}
-		middlewaresByIP[ip] = m
+
+		key := common.GetUniqueIDFromUUID()
+		middlewares[key] = m
+		middlewaresByIP[ip] = key
+
+		if _, ipnet, err := net.ParseCIDR(ip); err == nil {
+			byCIDRs[ip] = ipnet
+		}
 	}
 
+	middlewareCache, _ := lru.New(40000)
```

Comment:
I have a question. Where is this cache size(40000) from?

Reply:
I don't have exact and proper value yet, but I guess `40000` will be enough, `40kb`.